python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py

```diff
@@ -0,0 +1,212 @@
+
+        logger.log(
+            log_level,
+            f"Creating manifest.json for {project.project_dir} using {self._generate_cli_args}.",
+        )
+        manifest_path = (
+            DbtCliResource(project_dir=os.fspath(project.project_dir))
+            .cli(
+                self._generate_cli_args,
+                target_path=project.target_dir,
+            )
+            .wait()
+            .target_path.joinpath("manifest.json")
+        )
+        logger.log(log_level, f"Created {manifest_path}.")
+        return manifest_path
+
+
+@experimental
+class DbtProject(DagsterModel):
+    project_dir: Path
+    target_dir: Path
+    manifest_path: Path
+    packaged_project_dir: Optional[Path]
+    manifest_handler: DbtManifestHandler
+
+    def __init__(
+        self,
+        project_dir: Union[Path, str],
+        *,
+        target_dir: Union[Path, str] = "target",
+        packaged_project_dir: Optional[Union[Path, str]] = None,
+        manifest_handler: DbtManifestHandler = DagsterDbtManifestHandler(),
+    ):
+        """Representation of a dbt project.
+
+        Args:
+            project_path (Union[str, Path]):
+                The directory of the dbt project.
+            target_dir (Union[str, Path]):
+                The folder in the project directory to output artifacts.
+                Default: "target"
+            manifest_handler (Optional[DbtManifestHandler]):
+                A handler which receives lifecycle callbacks to ensure manifest.json
+                is in the right state at the right times.
+                Default: DagsterDbtManifestHandler
+            packaged_project_dir (Optional[Union[str, Path]]):
+                A directory that will contain a copy of the dbt project and the manifest.json
+                when the artifacts have been built. The prepare method will handle syncing
+                the project_path to this directory.
+                This is useful when the dbt project needs to be part of the python package data
+                like when deploying using PEX.
+        """
+        project_dir = Path(project_dir)
+        if not project_dir.exists():
+            raise DagsterDbtProjectNotFoundError(f"project_dir {project_dir} does not exist.")
+
+        packaged_project_dir = Path(packaged_project_dir) if packaged_project_dir else None
+        if using_dagster_dev() and packaged_project_dir:
+            current_project_dir = packaged_project_dir
+        else:
+            current_project_dir = project_dir
+
+        manifest_path = current_project_dir.joinpath(target_dir, "manifest.json")
+
+        super().__init__(
+            project_dir=project_dir,
+            target_dir=target_dir,
+            manifest_path=manifest_path,
+            packaged_project_dir=packaged_project_dir,
+            manifest_handler=manifest_handler,
+        )
+        manifest_handler.on_load(self)
+
+
+@experimental
+class DagsterDbtProjectManager:
```

Comment:
I think the factoring of this out is good, but the name is meh to me. It could be more deployment specific but `DagsterDbtProjectDeployer`  /  `DagsterDbtProjectPreparer` don't feel great

Reply:
FWIW I was going to suggest `DagsterDbtProjectDeployer` before I saw this comment.  Although I guess an issue with that is that it's not carrying out the full deployment process, just doing some preparatory steps? `DbtProjectDeploymentPreparer`? Would it be easier if it was just a utility function instead of a class?

Reply:
My "don't feel great" comment was more so just they did not feel good to say out loud than a logical position. They don't seem as bad with fresh eyes this morning, and the fact they seemed intuitive to you is another positive signal. I will try out functions to see how that is.